src/components/entry/EntryDetails.js

```diff
@@ -1,19 +1,204 @@
+import { Flag, Map, Star } from '@styled-icons/boxicons-solid'
+import React, { useEffect, useState } from 'react'
 import { useRouteMatch } from 'react-router-dom'
+import styled from 'styled-components'
 
-const EntryDetails = () => {
+import { getLocationById, getTypeById } from '../../utils/api'
+import Button from '../ui/Button'
+import { theme } from '../ui/GlobalStyle'
+import IconButton from '../ui/IconButton'
+import { Tag, TagList } from '../ui/Tag'
+import { RESOURCES } from './resources'
+
+const ACCESS_TYPE = {
+  0: "On lister's property",
+  1: 'Received permission from owner',
+  2: 'Public property',
+  3: 'Private but overhanging',
+  4: 'Private property',
+}
+
+const formatISOString = (dateString) =>
+  new Date(dateString).toLocaleDateString(undefined, {
+    year: 'numeric',
+    month: 'long',
+    day: 'numeric',
+  })
+
+// TODO: Reduce number of styled components by using selectors in the container
+
+// Wraps the entire page and gives it a top margin if on mobile
+const EntryDetailsPageContainer = styled.div`
+  margin-top: ${(props) => (props.isDesktop ? '0px' : '90px')};
+`
+
+const ImageContainer = styled.img`
+  width: 100%;
+`
+
+// Wraps all text in the container
+const EntryDetailsContent = styled.div`
+  padding: 23px;
+  box-sizing: border-box;
+`
+
+const PlantName = styled.h2`
+  margin-top: 0px;
+  margin-bottom: 0px;
+  color: #333333;
+`
+
+const ScientificName = styled.small`
+  font-style: italic;
+`
+
+// Wraps the plant name and scientific name, as well as an icon button for desktop
+const HeaderContainer = styled.header`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`
+
+// Wraps description, last updated text, and review and report buttons
+const DescriptionContainer = styled.section`
+  & > *:not(:last-child) {
+    margin-bottom: 14px;
+  }
+`
+
+const Description = styled.p`
+  color: #5a5a5a;
+`
+
+const UpdateText = styled.p`
+  font-size: 14px;
+  font-style: italic;
+`
+
+const ButtonSpacing = styled.div`
+  button {
+    margin-right: 14px;
+  }
+`
+
+const ResourceHeader = styled.h3`
+  color: #333333;
+`
+
+const ResourceLink = styled.a`
+  font-size: 16px;
+`
+
+const ResourceImage = styled.img`
+  height: 20px;
+  width: 25px;
+`
+
+// Wraps all resource images and their links
+const IndividualResourceContainer = styled.small`
+  & > *:not(:last-child) {
+    margin-bottom: 14px;
+  }
+  display: flex;
+  column-count: 2;
+  column-gap: 12px;
+`
+
+const EntryDetails = ({ isDesktop }) => {
   const {
     params: { id },
   } = useRouteMatch()
 
-  return (
-    <div>
-      <br />
-      <br />
-      <br />
-      <br />
-      <br />
-      <p>EntryDetails for id: {id}</p>
-    </div>
+  const [locationData, setLocationData] = useState()
+  const [locationTypeData, setLocationTypeData] = useState()
+
+  useEffect(() => {
+    async function fetchEntryDetails() {
```

Comment:
can this be an arrow function?

Reply:
They are the same in this case. I was thinking about how we should standardize arrow vs. standard functions throughout the repo (when to use which one). Would like to get everyone's thoughts on this. For now, it should be ok.